tealiumlibrary/src/main/java/com/tealium/core/Tealium.kt

```diff
@@ -316,6 +317,57 @@ class Tealium @JvmOverloads constructor(val key: String, val config: TealiumConf
      */
     @Suppress("unused")
     fun killTraceVisitorSession() {
-        activityObserverListener.killTraceVisitorSession()
+        deepLinkHandler.killTraceVisitorSession()
+    }
+
+    /**
+     * Migrates persistent data from the Tealium Android (Java) library if present
+     * */
+    private fun migratePersistentData() {
+        val hashCode = (config.accountName + '.' +
+                    config.profileName + '.' +
+                    config.environment.environment).hashCode()
+        val legacySharedPreferences = config.application.getSharedPreferences("tealium.datasources.${Integer.toHexString(hashCode)}", 0)
+        if (legacySharedPreferences.all.isEmpty()) {
+            return
+        }
+        legacySharedPreferences.all.forEach { item ->
+            val key = item.key
+            val value = item.value
+
+            (value as? String)?.let {
```

Comment:
I'd recommend this gets changed to a `when` statement as it'll look much cleaner - something like: 
```
when (value) {
        is String -> dataLayer.putString(key, value, Expiry.FOREVER)
        /* .. others .. */
        is Set<*> -> {
            val list = value.filterIsInstance<String>()
            dataLayer.putStringArray(key, list.toTypedArray(), Expiry.FOREVER)
        }
    }
```